web/docker/entrypoint.sh

```diff
@@ -1,29 +1,41 @@
 #!/bin/bash
 
-USER_ID=$($USER)
+USER_ID="$(id -u)"
 USER_GROUP="$(id -g)"
 WORKSPACE_DIR="/workspace"
 PGPASS_FILE=$WORKSPACE_DIR/.pgpass
 
-if [ "$(id -u)" = '0' ]; then
-  # Create workspace directory and change the permission to 'codechecker' user
-  # if this directory doesn't exists.
-  if [ ! -d $WORKSPACE_DIR ]; then
+if [ "$USER_ID" = '0' ]; then
+  echo "Container started with 'root' user."
+
+  if [ -d $WORKSPACE_DIR ]; then
+    echo "Workspace directory '${WORKSPACE_DIR}' already exists."
+
+    workspace_dir_owner=$(stat ${WORKSPACE_DIR} -c %u)
+    if [ "${workspace_dir_owner}" != '0' ]; then
+      echo "Executing script with workspace directory owner (UID): '${workspace_dir_owner}'..."
+      exec gosu $workspace_dir_owner "$0" "$@"
+    fi
+  else
+    echo "Creating workspace directory: '${WORKSPACE_DIR}'."
+    echo "WARNING: This directory exists ONLY within the containter!"
+
     mkdir -p $WORKSPACE_DIR
     chown -R codechecker:codechecker $WORKSPACE_DIR
-  fi
 
-  # Execute this script again with codechecker user.
-  exec gosu codechecker "$0" "$@"
+    echo "Executing script with internal 'codechecker' user (UID: $USER_ID)..."
+    exec gosu codechecker "$0" "$@"
```

Comment:
Does this exec do something meaningful? `$0` should start the `entrypoint.sh` script again, no?

I was expecting the following output:

```
Started with root
Creating dir
Executing with internal user ...
(self-call begins here)
Started with internal user
Executing command CodeChecker blah...
```

But instead, I got this:

```
$ docker run codechecker-web:csordasmarton/fix_docker_entry_point
Container started with 'root' user.
Creating workspace directory: '/workspace'.
WARNING: This directory exists ONLY within the containter!
Executing script with internal 'codechecker' user (UID: 0)...
Executing command: 'CodeChecker server --workspace /workspace --not-host-only'.
[INFO 2021-12-08 11:20] - Checking configuration database ...
```

But the process is running with the right userid (950).

Reply:
Yes, we will start the enrypoint again. In the second time what we will print is only the `Executing command: ...` message and the logs of the started process. Other logs are printed when the container is started with the root user.

Reply:
Ah... gotcha! Yeah because the `Container started with 'root' user.` is hardcoded and printed ONLY if we are root, it is not a generic message that prints the current username in *every* case.

Alright.